py_hamt/hamt.py

```
@@ -197,6 +188,32 @@ def __init__(self, hamt: "HAMT") -> None:
         self.hamt: HAMT = hamt
         # The integer key is a uuidv4 128-bit integer, for (almost perfectly) guaranteeing uniqueness
         self.buffer: dict[int, Node] = {}
+        # CAS IDs identify immutable data, so clean nodes can be reused until the
+        # cache is explicitly vacated. Dirty nodes remain exclusively in buffer.
+        self.clean_cache: dict[IPLDKind, Node] = {}
+
+    def get_clean_node(self, id: IPLDKind) -> Node | None:
```

Ruff (0.15.20)

[error] Function argument `id` is shadowing a Python builtin

(A002)